components/public-api/gitpod/experimental/v2/organization.proto

```diff
@@ -0,0 +1,208 @@
+}
+
+message OrganizationMember {
+    string user_id = 1;
+    OrganizationRole role = 2;
+    google.protobuf.Timestamp member_since = 3;
+    optional string avatar_url = 4;
+    optional string full_name = 5;
+    optional string primary_email = 6;
+    bool owned_by_organization = 7;
+}
+
+enum OrganizationRole {
+    ORGANIZATION_ROLE_UNSPECIFIED = 0;
+    ORGANIZATION_ROLE_OWNER = 1;
+    ORGANIZATION_ROLE_MEMBER = 2;
+}
+
+message OrganizationSettings {
+    bool workspace_sharing_disabled = 1;
+    optional string default_workspace_image = 2;
+}
+
+service OrganizationService {
+    // CreateOrganization creates a new Organization.
+    rpc CreateOrganization(CreateOrganizationRequest) returns (CreateOrganizationResponse) {};
+
+    // GetOrganization retrieves a single Organization.
+    rpc GetOrganization(GetOrganizationRequest) returns (GetOrganizationResponse) {}
+
+    // UpdateOrganization updates the properties of an Organization.
+    rpc UpdateOrganization(UpdateOrganizationRequest) returns (UpdateOrganizationResponse) {};
+
+    // ListOrganizations lists all organization the caller has access to.
+    rpc ListOrganizations(ListOrganizationsRequest) returns (ListOrganizationsResponse) {};
+
+    // DeleteOrganization deletes the specified organization.
+    rpc DeleteOrganization(DeleteOrganizationRequest) returns (DeleteOrganizationResponse) {};
+
+    // GetOrganizationInvitation retrieves the invitation for a Organization.
+    rpc GetOrganizationInvitation(GetOrganizationInvitationRequest) returns (GetOrganizationInvitationResponse) {};
+
+    // JoinOrganization makes the caller a OrganizationMember of the Organization.
+    rpc JoinOrganization(JoinOrganizationRequest) returns (JoinOrganizationResponse) {};
+
+    // ResetOrganizationInvitation resets the invitation_id for a Organization.
+    rpc ResetOrganizationInvitation(ResetOrganizationInvitationRequest) returns (ResetOrganizationInvitationResponse) {};
+
+    // ListOrganizationMembers lists the members of a Organization.
+    rpc ListOrganizationMembers(ListOrganizationMembersRequest) returns (ListOrganizationMembersResponse) {};
+
+    // UpdateOrganizationMember updates organization membership properties.
+    rpc UpdateOrganizationMember(UpdateOrganizationMemberRequest) returns (UpdateOrganizationMemberResponse) {};
+
+    // DeleteOrganizationMember removes a OrganizationMember from the Organization.
+    rpc DeleteOrganizationMember(DeleteOrganizationMemberRequest) returns (DeleteOrganizationMemberResponse) {};
+
+    // GetOrganizationSettings retrieves the settings of a Organization.
+    rpc GetOrganizationSettings(GetOrganizationSettingsRequest) returns (GetOrganizationSettingsResponse) {};
+
+    // UpdateOrganizationSettings updates the settings of a Organization.
+    rpc UpdateOrganizationSettings(UpdateOrganizationSettingsRequest) returns (UpdateOrganizationSettingsResponse) {};
+}
+
+message UpdateOrganizationRequest {
```

Comment:
Otherwise I'd find the usage confusing.

```suggestion
message UpdateOrganiuationNameRequest {
```

Reply:
We are updating the organization which currently only has one mutable field.